File: #0377.combination-sum-iv.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {

        vector<unsigned int> result(target + 1, 0);
        result[0] = 1;
        for (int i = 1; i <= target; i++) {
            for (int num : nums) {
                if (num <= i) {
                    result[i] += result[i - num];
                }
            }
        }
        return result[target];
    }
};

File: #0377.combination-sum-iv.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var combinationSum4 = function(nums, target) {
    let result = new Array(target + 1).fill(0);
    result[0] = 1;
    for (let i = 1; i < target + 1; i++) {
        for (let j = 0; j < nums.length; j++) {
            if (nums[j] <= i) {
                result[i] += result[i - nums[j]];
            }
        }
    }
    return result[target];
};
